### PR TITLE
Make 'ID' optional in a WorkItem attribute

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@jcouv ptal.

**Comment by jaredpar** [comment]
Way back in the days of yore, the URLs for all the various work item types were not so easily parsable. This is why the ID field was added, to let analysis happen on the code base without having to encode 5-10 ways of parsing tricky URLs to find the bug number in question. That was back in the days of yore though, these days everything is easily parsable.

File: src/Compilers/Test/Core/Assert/WorkItemAttribute.cs [hunk 1]
```diff
@@ -36,5 +36,14 @@ public WorkItemAttribute(int id, string issueUri)
             Id = id;
             Location = issueUri;
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WorkItemAttribute"/>.
+        /// </summary>
+        /// <param name="issueUri">The URI where the work item can be viewed. This is a link to work item in the
+        /// original source.</param>
+        public WorkItemAttribute(string issueUri) : this(-1, issueUri)
```
[the remark this answers is not in the document]
Comment (jaredpar) [reply src/Compilers/Test/Core/Assert/WorkItemAttribute.cs:45]: Oh did we turn off your other PR to just remove all the instances where we passed the ID?

[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Compilers/Test/Core/Assert/WorkItemAttribute.cs:45]: Yes. It was rejected :)

**Comment by CyrusNajmabadi** [comment]
> Why even keep around the other ctor?

One reason si that we have a handful of WorkItems where the id doesn't match between the number and the string.  And there are a small handful where the uri is just something like `"DevDiv"` :)  

I'd love to go clean all those up as well, but that's a more onerous task, hunting down what those all mean.  For now though, this allows the 99.9% case to migrate.  My intent is to change teh IDE to use this universally.  Compiler will stay with the existing redundant form for now.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1)

**Comment by AlekseyTs** [comment]
Just a reminder, test only changes can be merged with a single sign-off from the Compiler team.

**Comment by CyrusNajmabadi** [comment]
> Just a reminder, test only changes can be merged with a single sign-off from the Compiler team.

Sounds good.  Thanks!

**Comment by jaredpar** [comment]
@CyrusNajmabadi 

> One reason si that we have a handful of WorkItems where the id doesn't match between the number and the string. And there are a small handful where the uri is just something like "DevDiv" :)

Once you merge the other change I'll go through and clean up the rest. The thought being that if I don't have context for whath the right number is probably no one does so can fix or delete.

**Comment by CyrusNajmabadi** [comment]
> The thought being that if I don't have context for whath the right number is probably no one does so can fix or delete.

I def think it can be figured out.  For example, if we have `WorkItem(1, "...2...")` just go look in the db for both of those and see which applies :)

But i def appreciate any help here to make this hte consistent pattern we use for everything.  it will help make things much cleaner and simpler (and will avoid this exact issue which clearly happened to a bunch of tests :)).

**Pull request merged**